### PR TITLE
Issue #2504025: Support Drupal 8 plugins via annotation parsing

### PR DESCRIPTION
**Pull request by drupol**
This is a first draft of a new submodule.

It allows you to use Drupal 8 annotations for your plugins (see doc: https://www.drupal.org/node/1882526)

### CODE REVIEW

File: lib/Drupal/Core/DependencyInjection/Dumper/PhpArrayDumper.php [hunk 1]
```diff
@@ -84,7 +84,8 @@ protected function getServiceDefinitions()
       while (isset($aliases[(string) $id])) {
         $id = $aliases[(string) $id];
       }
-      $services[$alias] = (string) $this->getServiceAliasDefinition($id);
+      $alias_definition = $this->getServiceAliasDefinition($id);
```
Comment (drupol) [inline lib/Drupal/Core/DependencyInjection/Dumper/PhpArrayDumper.php:87]: Oops, I committed this by error. I mean, this is part of another pull request: #61

**Comment by drupol** [comment]
Doctrine only available for PHP > 5.4 : http://doctrine-orm.readthedocs.org/en/latest/tutorials/getting-started.html

**Comment by dawehner** [comment]
There has to be a version with support for it ...

@fabian
What is the reason that we require such an old PHP version?

**Comment by drupol** [comment]
Actually, the problem comes from here: https://github.com/LionsAd/service_container/blob/7.x-1.x/lib/Drupal/Component/FileCache/FileCacheFactory.php#L41
I'll see what I can do.

**Comment by drupol** [comment]
I'm tempted to make a local change, but I don't like that.

**Comment by drupol** [comment]
See #66

**Pull request closed without merging**